Content.Shared/StepTrigger/StepTriggerSystem.cs

```diff
@@ -18,62 +18,62 @@ public override void Initialize()
 
     public override void Update(float frameTime)
     {
-        foreach (var (_, trigger) in EntityQuery<StepTriggerActiveComponent, StepTriggerComponent>())
+        foreach (var (_, trigger, transform) in EntityQuery<StepTriggerActiveComponent, StepTriggerComponent, TransformComponent>())
         {
-            if (!Update(trigger))
+            if (!Update(trigger, transform))
                 continue;
 
             RemComp<StepTriggerActiveComponent>(trigger.Owner);
         }
     }
 
-    private bool Update(StepTriggerComponent component)
+    private bool Update(StepTriggerComponent component, TransformComponent transform)
     {
-        if (component.Deleted || !component.Active || component.Colliding.Count == 0)
+        if (component.Deleted ||
+            !component.Active ||
+            component.Colliding.Count == 0)
             return true;
 
+
         foreach (var otherUid in component.Colliding.ToArray())
         {
-            if (!otherUid.IsValid())
-            {
-                component.Colliding.Remove(otherUid);
-                component.CurrentlySteppedOn.Remove(otherUid);
-                component.Dirty();
-                continue;
-            }
-
-            // TODO: This shouldn't be calculating based on world AABBs.
-            var ourAabb = _entityLookup.GetWorldAABB(component.Owner);
-            var otherAabb = _entityLookup.GetWorldAABB(otherUid);
+            var shouldRemoveFromColliding = UpdateColliding(component, transform, otherUid);
+            if (!shouldRemoveFromColliding) continue;
 
-            if (!TryComp(otherUid, out PhysicsComponent? otherPhysics) || !ourAabb.Intersects(otherAabb))
-            {
-                component.Colliding.Remove(otherUid);
-                component.CurrentlySteppedOn.Remove(otherUid);
-                component.Dirty();
-                continue;
-            }
+            component.Colliding.Remove(otherUid);
+            component.CurrentlySteppedOn.Remove(otherUid);
+            component.Dirty();
+        }
+        return false;
+    }
 
-            if (component.CurrentlySteppedOn.Contains(otherUid))
-                continue;
+    private bool UpdateColliding(StepTriggerComponent component, TransformComponent ownerTransform, EntityUid otherUid)
+    {
+        if (!otherUid.IsValid() ||
+            !TryComp(otherUid, out PhysicsComponent? otherPhysics))
```

Comment:
If the cause is steptrigger leading to other colliding entities being deleted then I'm fine with this here I guess.